### PR TITLE
Sonnet V1.2.14

### PR DESCRIPTION
**Pull request by ultrabear**
This version leaves a short but meaningful changelog full of user requested features and improvements, all new features have proven to be stable enough on the dev instance and I dont want to bloat the release too much

### CODE REVIEW

File: cmds/cmd_moderation.py [hunk 1]
```diff
@@ -164,7 +164,13 @@ class InfractionGenerationError(Exception):
     __slots__ = ()
 
 
-InfractionInfo = Tuple[Optional[discord.Member], InterfacedUser, str, str, Optional[Awaitable[None]]]
+class InfractionInfo(NamedTuple):
+    memeber: Optional[discord.Member]
```
Comment (bredo228) [inline cmds/cmd_moderation.py:168]: memeber?

**Review by bredo228 — APPROVED** [review]
looks good apart from that one typo

File: cmds/cmd_moderation.py (hunk 1 above)
Comment (ultrabear) [reply cmds/cmd_moderation.py:168]: oh god

Comment (ultrabear) [reply cmds/cmd_moderation.py:168]: image
i should really add codespell to ci

Comment (bredo228) [reply cmds/cmd_moderation.py:168]: probably a good idea

**Comment by bredo228** [comment]
all good to merge now

**Comment by ultrabear** [comment]
# Sonnet V1.2.14
## Frontend
- Add(mprotect): Moderator+ infraction protection tools
- Change(join-log): Join logs alone may now be disabled through `set-leave-log-is-join-log`
- Change(message-log): Message logs alone may now be disabled through `set-edit-log-is-message-log`
- Add(unmute-on-ban): Banned members may now be auto unmuted via `set-unmute-on-ban`
- Change(purge): Purge now prints how many messages were purged and what user initiated the purge
- Change(automod): Automod is now active on bot users, this is to allow use of webhook based user accounts
- Change(rr-add/remove): Adding or removing a reactionrole will make the bot attempt to add or remove a reaction of that emoji to the message
- Change(help): Help command no longer prints full command list if length of one embed field is larger than 512 bytes, instead truncates
- Change(version-info): Will now print the time of last reload to version-info, letting users track the last patch date of the sonnet instance
## Under the Hood
- Add(private_message): Privatized error message endpoint for CommandErrors
  - Currently this is unused but has an api endpoint so it can be used in the future
- Refactor(antispam): Antispam code has been completely rewritten, improving readability and stability substantially
- Refactor(on_message(_edit)): Systems no longer panic on missing ban/kick/mute/warn commands in automod, instead they warn to terminal
- Update(lexdpyk): release LeXdPyK 1.4.15: improve logging and UX, make all events logged in developer mode (replaces dev.py kernel)
- Change(parse_skip_message): Add flag to allow bot users for parsing anyways, while still skipping self
- Add(lib_lexdpyk_h): BotOwners class to lexdpyk headers to allow accessing the current bot owners
- Add(parse_boolean_strict): Parse boolean clone that has a more sane api
- Deprecate(parse_boolean): Deprecate parse boolean as the better method parse_boolean_strict is now available
- Add(manualtest): Add extra ramfs assertions to manualtest
- Add(boolean_to_db_helper): Administration subcommand to parse a boolean entry to the database
- Change(reactionroles): Reactionrole data is now stored in a new format on the database in a forward incompatible way
  - Downgrading sonnet 1.2.14 reactionrole data to 1.2.13-3 or below will not work as intended
- Change(filelog): Files in the file log are now stored for 60 minutes instead of 30 minutes
## Runtime Environment
- Add(cached_yapf): Build tool to autocache yapf results and speed up linting time
## Bugs
- Patch(allowed_chars): Spaces are now allowed in message contents before it is sent to the file logging system
- Patch(antispam): Previously sending 2 messages that should be antispammed between a lagspike would not trigger antispam, this is no longer the case
- Patch(reactionroles): Animated emojis will now work with reactionroles

**Comment by ultrabear** [comment]
using a combination of the github issues and git commits and reading all the files works really nice

**Pull request merged**